devtools/gradle/src/main/java/io/quarkus/gradle/dependency/ConditionalDependenciesEnabler.java

```diff
@@ -0,0 +1,193 @@
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+import org.gradle.api.GradleException;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.Dependency;
+import org.gradle.api.artifacts.ModuleVersionIdentifier;
+import org.gradle.api.artifacts.ResolvedArtifact;
+import org.gradle.api.artifacts.dsl.DependencyHandler;
+import org.gradle.api.plugins.JavaPlugin;
+
+import io.quarkus.bootstrap.BootstrapConstants;
+import io.quarkus.bootstrap.util.ZipUtils;
+
+public class ConditionalDependenciesEnabler {
+
+    private Map<String, ExtensionDependency> featureVariants = new HashMap();
+    private Project project;
+
+    public ConditionalDependenciesEnabler(Project project) {
+        this.project = project;
+    }
+
+    public void addConditionalDependencies() {
+        Configuration implementationCopy = DependencyUtils.duplicateConfiguration(project,
+                project.getConfigurations().getByName(JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME));
+        Set<ResolvedArtifact> runtimeArtifacts = implementationCopy.getResolvedConfiguration().getResolvedArtifacts();
+        List<ExtensionDependency> firstLevelExtensions = collectFirstLevelExtension(runtimeArtifacts,
+                project.getDependencies());
+        featureVariants.putAll(extractFeatureVariants(firstLevelExtensions));
+        resolveConditionalDependencies(project.getDependencies(), firstLevelExtensions, implementationCopy);
+    }
+
+    private List<ExtensionDependency> collectFirstLevelExtension(Set<ResolvedArtifact> runtimeArtifacts,
+            DependencyHandler dependencies) {
+        List<ExtensionDependency> firstLevelExtensions = new ArrayList<>();
+        for (ResolvedArtifact artifact : runtimeArtifacts) {
+            ExtensionDependency extensionDependency = getExtensionInfoOrNull(artifact, artifact.getFile(),
+                    dependencies);
+            if (extensionDependency != null) {
+                if (!extensionDependency.conditionalDependencies.isEmpty()) {
+                    if (extensionDependency.needsResolution(runtimeArtifacts)) {
+                        firstLevelExtensions.add(extensionDependency);
+                    }
+                }
+            }
+        }
+        return firstLevelExtensions;
+    }
+
+    private Map<String, ExtensionDependency> extractFeatureVariants(List<ExtensionDependency> extensions) {
+        Map<String, ExtensionDependency> possibleVariant = new HashMap<>();
+        for (ExtensionDependency extension : extensions) {
+            for (Dependency dependency : extension.conditionalDependencies) {
+                possibleVariant.put(DependencyUtils.asFeatureName(dependency), extension);
+            }
+        }
+        return possibleVariant;
+    }
+
+    private void resolveConditionalDependencies(DependencyHandler dependencies,
```

Comment:
`dependencies` seems to be always `project.getDependencies()`? Doesn't have to be passed as a method argument?